src/hotspot/cpu/aarch64/templateTable_aarch64.cpp

```diff
@@ -2467,9 +2531,11 @@ void TemplateTable::getfield_or_static(int byte_no, bool is_static, RewriteContr
   const Register raw_flags = r6;
   const Register bc    = r4; // uses same reg as obj, so don't mix them
 
-  resolve_cache_and_index(byte_no, cache, index, sizeof(u2));
+  resolve_cache_and_index_for_field(byte_no, cache, index);
   jvmti_post_field_access(cache, index, is_static, false);
-  load_field_cp_cache_entry(obj, cache, index, off, raw_flags, is_static);
+  load_resolved_field_entry(obj, cache, index, off, raw_flags, is_static);
+  // Index holds the TOS
+  __ mov(flags, index);
```

Comment:
This is very confusing. You call the same thing "TOS", "type" and "index". Please use consistent naming and make it more comprehensive. In addition, why do you need the extra move?

Reply:
Great point, this is very unclear. I will use better names which should remove the need for any moves.